### ISSUE TITLE
[glusterfs] collect /var/log/glusterfs/cmd_history.log

### ISSUE THREAD

**Issue by satheesaran** [sosreport/sos#484]
currently glusterfs plugin for sos doesn't collect this file - /var/log/glusterfs/cmd_history.log
It is expected to have this log file collected by sos by default. ( without using -a )

**Comment by bmr-cymru** [comment]
Could you provide some sample content for this file?

**Comment by satheesaran** [comment]
glusterfs plugin available with RHEL 6.6 is little different from the one available here with this sos utility.
I tried to install this sosreport and find it getting all the files. Now the change is required only in the sos script which is in downstream. 

Closing this issue

**Comment by bmr-cymru** [comment]
Changes need to go upstream first. Then they get backported to legacy branches (in this case [RHEL-6](https://github/com/sosreport/sos/tree/rhel-6)).

**Comment by satheesaran** [comment]
Thanks for that information.
The downstream sos available in RHEL 6.6, collects the specific files under /var/log/glusterfs by default.
This is done intentionally to avoid 'log-rotated' files in the sosreport collection.
So the downstream glusterfs.py has this code additionally added,

# all_logs takes precedence over logsize
        if not self.getOption("all_logs"):
            limit = self.getOption("logsize")
        else:
            limit = 0

        if limit:
            # collect logs last as some of the other actions create log entries
            self.addCopySpecLimit("/var/log/glusterfs/cli.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/*.vol.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/gluster-lock.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/glustershd.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/nfs.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/quota-crawl.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/quotad.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/quotad-mount-*.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/status.log", limit)
            self.addCopySpecLimit("/var/log/glusterfs/bricks/*.log", limit)
        else:
            self.addCopySpec("/var/log/glusterfs")

With the above code we also need to collect, /var/log/glusterfs/cmd_history.log
These changes needs to be added to gluster.py

**Comment by bmr-cymru** [comment]
That's not related to this issue - please post the sample "cmd_history.log" that was requested so that we can move this issue forward.

**Comment by satheesaran** [comment]
Bryn, 
Is my above comment  is relevant to your question ?

**Comment by bmr-cymru** [comment]
Thanks - that's what we needed to see. Is this file rotated by the process that creates it or is it managed with e.g. logrotate?

It's important to know this as we need to handle both the default size-limited log collection and 'all-logs' mode (where we collect everything).

**Comment by satheesaran** [comment]
Yes, this file is rotated by glusterfs.

[~]# cat /etc/logrotate.d/glusterfs                                                                                                                     
# Rotate client logs                                                                                                                                                    
/var/log/glusterfs/*.log {                                                                                                                                              
  sharedscripts                                                                                                                                                         
  weekly                                                                                                                                                                
  rotate 52                                                                                                                                                             
  missingok                                                                                                                                                             
  compress                                                                                                                                                              
  delaycompress                                                                                                                                                                                                
  notifempty                                                                                                                                                                                                   
  postrotate                                                                                                                                                                                                   
  /usr/bin/killall -HUP glusterfs > /dev/null 2>&1 || true                                                                                                                                                     
  /usr/bin/killall -HUP glusterd > /dev/null 2>&1 || true                                                                                                                                                      
  endscript                                                                                                                                                                                                    
}                                                                                                                                                                                                              
                                                                                                                                                                                                               
# Rotate server logs                                                                                                                                                                                           
/var/log/glusterfs/bricks/*.log {                                                                                                                                                                              
  sharedscripts                                                                                                                                                                                                
  weekly                                                                                                                                                                                                       
  rotate 52                                                                                                                                                                                                    
  missingok                                                                                                                                                                                                    
  compress                                                                                                                                                                                                     
  delaycompress                                                                                                                                                                                                
  notifempty                                                                                                                                                                                                   
  postrotate                                                                                                                                                                                                   
  /usr/bin/killall -HUP glusterfsd > /dev/null 2>&1 || true                                                                                                                                                    
  endscript
}

**Comment by bmr-cymru** [comment]
This should probably be blocked by #502 - there's a number of changes in the downstream plugin (on branch rhel-6) that need to be merged into master.

**Comment by TurboTurtle** [comment]
This log location would be collected by the current plugin per https://github.com/sosreport/sos/blob/master/sos/plugins/gluster.py#L77

Closing this issue.
